### ISSUE TITLE
News Section 

### ISSUE THREAD

**Issue by samu-wmde** [codingdavinci/relaunch2018#4]
As a founder I want to control which news articles appear as teasers on the landing page so that I can balance the kind of news we are highlighting. 

Add more features you would expect of a normal chronological blog: 
- categories
- tags
- Currently, we cut the number of accessible news articles to the most recent 7 - this is an oversight - should be able to access all past articles, chronologically listed over multiple pages.
- Optimise use of space, especially on mobile.

**Comment by Snater** [comment]
The news initially displayed will depend on a checkbox ("Promoted to front page") that may be ticked on each news article by a "Founder" user. Additional default filter options are to be defined in order to not have to un-tick that checkbox for articles that should not show up on the front page any more - i.e. show only the <number> most recent articles that are promoted to front page - that was the solution I noted down, but additional filters may also be added - i.e. show only the most recent article of each regional team promoted to front page.

The site visitor may apply individual filters according to the interaction design, i.e. tags, which will show all articles of that tag, or whatever filter is applied.

**Comment by lohengruen** [comment]
Here is the link to the screen design for a single news article (desktop and mobile)

https://www.figma.com/file/QWZaTBUKLoc4oGIM5eaF8L/News-Article-Design?node-id=0%3A1

**Comment by Snater** [comment]
@lohengruen On the News Overview page, the news items expand on click. How do I get to a single news item page?

**Comment by lohengruen** [comment]
also, on the single article page, we probably need a maximum height on the header image. If someone uploads a square image for example, it should be cropped on the upper und lower sides to be displayed in the format of ~16:9.

**Comment by Snater** [comment]
The only thing that seems to be tricky is the related articles. First off, I implemented automatically filling this list by retrieving the three most recent articles, excluding the current article, that have at least one tag in common with the current article.
Aligning the related article images to the bottom and and combining that hover effect without having a broad white space above the image is very tricky, as I do not want to specify a fixed height for the text content below the image. I left the hover effect out for now.

**Comment by Snater** [comment]
@lohengruen 
- Are the news article and news overview page headers final or do they depend on the key visuals? (The CdV logo on the article page features that Archive font. Shall I implement that as a graphic? Might be tricky to make it have the same height than the menu button.)
- How would the news overview header look like when filtered by a tag?
- In the news overview, the length of the article teaser texts are different. Shall the content editor supply a dedicated teaser text or shall the whole article text be stripped down to a fixed amount of characters for the teaser? (There is no such thing like a teaser break in Drupal and such could not be applied, since there is no particular single "body" field for the page. For the sake of flexibility, we are using the Paragraphs module to aggregate the content for the article.) I could also implement some sophisticated logic here (like take some minimum amount of characters and then continue to the paragraph end), but that'd probably be prone to errors.

**Comment by Snater** [comment]
OK, noticed the updated screen design.
Are those news pages headers final now?

**Comment by lucyWMDE** [comment]
re teaser length - I would be for a fixed amount of characters to keep it simple... but perhaps it depends on the design... @lohengruen ?

**Comment by lohengruen** [comment]
(sorry if this posting shows up two times - I've written it an answer some time ago but just noticed that it apparently didn't save my comment)

• headers: The individual article headers are final. Here the article image replaces the key visuals. On the overview news page, there will probably be some changes to the header, but the dimensions and the text content to the left will stay the same.
The text-logo is actually a combination of two fonts; the brackets are from Barlow, the letters from Archive. It's probably best to use a svg for the logo:
https://www.dropbox.com/s/d3nedo3ozdb58jd/CDV_Logo_Line.svg?dl=0

• related articles: if this causes too many problems, we can use a different hover effect, e.g. just a background color change. 

• teaser texts: If its no bother, I would opt for the solution with the dedicated intro text field, so that editors can control where there will be a break. I think it looks nicer if the articles in the overview have different teaser lengths (as well as different teaser image sizes). However, it's not something that absolutely needs to be that way, I would also be ok with a fixed character length for the teaser.

**Comment by lucyWMDE** [comment]
Looking at the single news article as it's been implemented, a couple of issues:
- should there be a max header image width (should there actually be a max page width for the whole content? seems some other pages have issues here... eg presse).
- In the screen design thumbnail images on the news overview page were different sizes - how should this be implemented?
@lohengruen

**Comment by lohengruen** [comment]
• there is a maximum width to the whole layout, which is around 1200px. The main wrapper is centered and everything to the left and right will be white space on large displays.
The news article header image is actually a bit smaller than the main wrapper: 1150px. There should be a white border around the image, that helps to distinguish it from the full width headers we use on other pages. The white border scales down with the image and disappears only on mobile devices. I've updated the screen design with a tablet version to illustrate this:
https://www.figma.com/file/QWZaTBUKLoc4oGIM5eaF8L/News-Article-Design?node-id=0%3A1

• On the news overview, I used different image sizes because I don't want to crop them in case theres information lost (e.g. a photo with the winners with one person being cut off). So the images should be roughly the same size, but with different aspect ratios. It could work with a combination of max-width and responsive max-height:
`img {
  max-width: 100%;
  height: auto;
  max-height: 20vh;
}`
This is of course just a suggestion. There may be smarter ways to do this.
